packages/neon/neon/lib/src/blocs/timer.dart

```diff
@@ -0,0 +1,72 @@
+part of '../../neon.dart';
+
+abstract class TimerBlocEvents {
+  /// Register a [callback] that will be called periodically.
+  /// The time between the executions is defined by the [duration].
+  TimerBlocTimer registerTimer(final Duration duration, final VoidCallback callback);
+
+  /// Unregister a timer that has been previously registered with the bloc.
+  /// You can also use [TimerBlocTimer.cancel].
+  void unregisterTimer(final TimerBlocTimer timer);
+}
+
+abstract class TimerBlocStates {}
+
+/// Execute callbacks at defined periodic intervals.
+/// Components can register their callbacks and everything with the same periodicity will be executed at the same time.
+///
+/// The [TimerBloc] is a singleton.
+/// Sub-second timers are not supported.
+class TimerBloc extends Bloc implements TimerBlocEvents, TimerBlocStates {
+  factory TimerBloc() => _singleton;
+  TimerBloc._();
+
+  static final TimerBloc _singleton = TimerBloc._();
+
+  final Map<int, Timer> _timers = {};
+  final Map<int, List<VoidCallback>> _callbacks = {};
+
+  @override
+  void dispose() {
+    for (final timer in _timers.values) {
+      timer.cancel();
```

Comment:
While this will cancel all timers they won't be removed from `_timers` or `_callbacks`.
We might want to clear them aswell (or just reassign the TimerBloc._singleton with a new instance).

Please also add unit tests for that case so this wont happen again.

Reply:
Since it's a singletone disposing doesn't make much sense, right?